src/Entity/BannerSelection/Campaign.php

```diff
@@ -12,24 +14,37 @@ class Campaign {
 	private $identifier;
 	private $start;
 	private $end;
+	private $rng;
+	private $displayRatio;
 
 	/**
 	 * @var Bucket[]
 	 */
 	private $buckets;
 
-	public function __construct( string $identifier, \DateTime $start, \DateTime $end, array $buckets ) {
+	public function __construct(
+		string $identifier,
+		\DateTime $start,
+		\DateTime $end,
+		float $displayRatio,
```

Comment:
Why are we passing in the ratio as a float? I guess it's because in our banner code we're using number between 0 and 1. Or is it because traditional math notation gives probabilities between 0 and 1?

Wouldn't it be easier if we used values betwen 0 and 100? 
- We could remove the math in `selectBucket`
- We would not have to do mental calculations when writing the config file

No hard requirement, just a random thought on how to improve our domain language.

What do you think, @tzhelyazkova ?

Reply:
I'm in favor of integers as long as we somehow explicitly state this is a percentage. Perhaps renaming the variable to `$displayPercentage` is the way to go because now that I think about it this is not a ratio.